assets/js/blocks/cart-checkout/checkout/sidebar/index.js

```diff
@@ -47,13 +50,15 @@ const CheckoutSidebar = ( {
 				removeCoupon={ removeCoupon }
 				values={ cartTotals }
 			/>
-			<TotalsShippingItem
-				currency={ totalsCurrency }
-				noResultsMessage={ null }
-				isCheckout={ true }
-				showCalculator={ false }
-				values={ cartTotals }
-			/>
+			{ needsShipping && (
+				<TotalsShippingItem
+					currency={ totalsCurrency }
+					noResultsMessage={ null }
+					isCheckout={ true }
```

Comment:
Anyplace we have an `isCheckout` prop can now be removed as we have a `isCart` flag returned from the `useCheckoutContext` hook (which will automatically be true if in the `CartProvider` tree and false if in the `CheckoutProvider` tree). No need to handle in this pull though.